src/smith/git.rs

```diff
@@ -177,7 +177,7 @@ impl Smith for Git {
             .find_reference("FETCH_HEAD")
             .into_report()
             .change_context(GitError::GitError)
-            .attach_printable_lazy(|| format!("Failed to find FETCH_HEAD: {url}"))
+            .attach_printable_lazy(|| format!("Failed to find FETCH_HEAD: {url}. This is most likely because the reference does not exist."))
```

Comment:
I'd probably change this error message, most people don't know what a reference is

Reply:
Something like "it is most likely because your git commit/tag/branch doesn't exist"?

Reply:
I think "Check if the specified commit, tag, or branch exists" sounds good